server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeon.java

```diff
@@ -63,7 +56,7 @@
  * of the same or different methods.
  */
 @Deprecated
-public class CuratorLoadQueuePeon extends LoadQueuePeon
+public class CuratorLoadQueuePeon implements LoadQueuePeon
```

Comment:
Change summary:
Commoned out class `SegmentHolder` as `QueuedSegment` to represent an item in a load queue and to be used by both `HttpLoadQueuePeon` and `CuratorLoadQueuePeon`.
Implemented new methods in `LoadQueuePeon`.